team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							const moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGenBreeding && (dexEntry.tier.startsWith('Bank') || moveGen !== '7' || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
> The moveGen stuff is, as mentioned, entirely unnecessary.

I don't think it is though. What I'm doing with `moveGen` is checking if the Pokemon we are trying to inherit the move from can learn that move in Gen 7. This means that `moveGen` and `learnedGen` won't always be the same, even if `learnedGen` is guaranteed to be 7.

> Not to mention that we don't use indexOf that way anymore

Sure, I'll change that.

> Also, template.id === dexEntry.prevo || dexEntry.evos.includes(template.id) should be unnecessary

I agree, and that's why i skipped this verification initially. However I was still strangely able to inherit a move from evos, and that's why I ended up adding those conditions.